File: core-android/src/main/java/com/lookaround/core/android/camera/IRenderSurface.java
package com.lookaround.core.android.camera;

import android.view.View;
import android.view.ViewStub;

import androidx.annotation.NonNull;

import com.google.common.util.concurrent.ListenableFuture;

interface IRenderSurface {
    @NonNull
    ListenableFuture<Void> waitForNextFrame();

    @NonNull
    View inflateWith(ViewStub viewStub, OpenGLRenderer renderer);
}

File: core-android/src/main/java/com/lookaround/core/android/camera/IRenderSurface.kt
package com.lookaround.core.android.camera

import android.view.View
import android.view.ViewStub
import com.google.common.util.concurrent.ListenableFuture

internal interface IRenderSurface {
    fun waitForNextFrame(): ListenableFuture<Unit>
    fun inflateWith(viewStub: ViewStub, renderer: OpenGLRenderer): View
}
